### PR TITLE
Modify reconciliation info endpoint

### PR DESCRIPTION
**Pull request by dbadura**
Proposed changes:
- modify OpenAPI
- remove cluster metadata from response
- adjust response to be aligned to endpoint functionality

See: https://github.com/kyma-project/control-plane/issues/1083

### CODE REVIEW

**Review by tobiscr — DISMISSED** [review]

File: cmd/mothership/mothership/start/helpers.go [hunk 1]
```diff
@@ -19,40 +18,3 @@ func components(cfg model.ClusterConfigurationEntity) []keb.Component {
 	}
 	return components
 }
-
-func clusterMetadata(runtimeID string, lastState *cluster.State) *keb.Cluster {
```
Comment (kwiatekus) [inline cmd/mothership/mothership/start/helpers.go:23]: We should not drop the whole `cluster` part of the response.. but just the kubeconfig.
See https://github.com/kyma-incubator/reconciler/issues/433, where @tobiscr expects non empty `components` list as part of that object which is here being removed

**Review by kwiatekus — CHANGES_REQUESTED** [review]

File: openapi/external_api.yaml [hunk 2]
```diff
@@ -333,11 +362,24 @@ components:
...
           format: uri
 
-    HTTPReconciliationOperations:
+    HTTPReconciliationInfo:
       type: object
+      required: [ lock, runtimeID, schedulingID, created, updated, status,operations ]
       properties:
-        cluster:
-          $ref: "#/components/schemas/cluster"
+        lock:
```
Comment (pPrecel) [inline openapi/external_api.yaml:369]: please remove this and add the `finished` field

File: openapi/external_api.yaml [hunk 3]
```diff
@@ -194,10 +194,29 @@ paths:
...
+            type: string
+            format: uuid
+        - name: version
+          required: true
+          in: path
+          schema:
+            type: string
+      responses:
+        "200":
+          $ref: "#/components/responses/configurationOkResponse"
```
Comment (pPrecel) [inline openapi/external_api.yaml:218]: One small suggestion:

```suggestion
          $ref: "#/components/responses/ConfigurationOkResponse"
```

File: openapi/external_api.yaml (hunk 2 above)
Comment (dbadura) [reply openapi/external_api.yaml]: Replaced

File: openapi/external_api.yaml (hunk 3 above)
Comment (dbadura) [reply openapi/external_api.yaml]: I will apply it in https://github.com/kyma-incubator/reconciler/pull/437

File: cmd/mothership/mothership/start/helpers.go (hunk 1 above)
Comment (dbadura) [reply cmd/mothership/mothership/start/helpers.go:23]: The components config will be implemented in this PR in another endpoint:
https://github.com/kyma-incubator/reconciler/pull/437

**Review by kwiatekus — APPROVED** [review]

**Pull request merged**